src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1605,13 +1619,56 @@ struct Apply<tmpl::list<Tags...>> {
  * \param args the arguments to pass to the function that are not in the
  * DataBox, `box`
  */
-template <typename TagsList, typename F, typename BoxTags, typename... Args>
+template <typename TagsList, typename F, typename BoxTags, typename... Args,
+          Requires<not DataBox_detail::has_argument_tags_v<std::decay_t<F>>> =
+              nullptr>
 inline constexpr auto apply(F&& f, const DataBox<BoxTags>& box,
-                            Args&&... args) {
+                            Args&&... args) noexcept {
   return DataBox_detail::Apply<TagsList>::apply(std::forward<F>(f), box,
                                                 std::forward<Args>(args)...);
 }
 
+template <
+    typename F, typename BoxTags, typename... Args,
+    Requires<DataBox_detail::has_argument_tags_v<std::decay_t<F>>> = nullptr,
+    typename ArgumentTags = typename std::decay_t<F>::argument_tags>
+inline constexpr auto apply(F&& f, const DataBox<BoxTags>& box,
+                            Args&&... args) noexcept {
+  return DataBox_detail::Apply<ArgumentTags>::apply(
+      std::forward<F>(f), box, std::forward<Args>(args)...);
+}
+// @}
+
+/*!
+ * \ingroup DataBoxGroup
+ * \brief Apply the stateless function `F::apply` that takes as arguments the
```

Comment:
How would this function be stateful? I'm confused what you mean by a `stateless function`